app/views/posts/_post.html.erb

```diff
@@ -1,3 +1,4 @@
+<% if friends?(post.user) || current_user == post.user %>
```

Comment:
As you can see, you are using if blocks to check if a post is made by a friend or current_user, which makes your view handle logic instead of presentation. Kindly use methods in your model to query specific post from the DB.